src/views/DashboardFiles.vue

```diff
@@ -39,6 +42,43 @@ function disconnect() {
 onMounted(() => {
   refreshFilesList()
 })
+const filteredFiles: ShallowRef<File[]> = shallowRef([])
+const title: ShallowRef<string> = shallowRef('All files')
+
+const listOfCategoriesSelected: Ref<string[]> = ref([])
+
+function findFile(searchText: string, categories: Chips[]) {
+
+  if(categories !== undefined){
+    listOfCategoriesSelected.value = categories
+      .filter((chips: Chips) => chips.clicked === true)
+      .map((chips: Chips) => chips.name)
+  }
+
+  if(searchText !== ''  && listOfCategoriesSelected.value.length) {
```

Comment:
split this huge if-statement. You compare `listOfCategoriesSelected.value.length` in all conditions.